torch/_functorch/aot_autograd.py

```diff
@@ -969,9 +970,10 @@ def inner(*flat_args):
             for o, info in zip(flat_f_outs, output_info)
             if info.output_type in [OutputType.non_alias, OutputType.unsafe_view_alias, OutputType.custom_function_view]
             and issubclass(info.raw_type, torch.Tensor)
+            and info.requires_grad
         ]
         # intermediate bases are also included in the backward graph
-        f_tangents = f_input_tangents + f_output_tangents + intermediate_bases
```

Comment:
hmm, I don't think we can't blindly remove `f_input_tangents` here, because if we have an input that was mutated **and** required gradients, then it needs to participate in the backward.

You should be able to filter `f_input_tangents` the same way you filtered `f_output_tangents` above:
```
        f_input_tangents = [
            inp
            for inp, info, requires_grad_info in zip(flat_f_args, input_info, input_requires_grad_info)
            if info.mutates_data and requires_grad_info
        ]
```